test/Fork/BaseSepoliaForkBase.t.sol

```diff
@@ -0,0 +1,123 @@
+import {NameResolver} from "ens-contracts/resolvers/profiles/NameResolver.sol";
+
+import {RegistrarController} from "src/L2/RegistrarController.sol";
+import {UpgradeableRegistrarController} from "src/L2/UpgradeableRegistrarController.sol";
+import {IL2ReverseRegistrar} from "src/L2/interface/IL2ReverseRegistrar.sol";
+import {IReverseRegistrar} from "src/L2/interface/IReverseRegistrar.sol";
+import {Sha3} from "src/lib/Sha3.sol";
+import {BASE_ETH_NODE} from "src/util/Constants.sol";
+
+import {BaseSepolia as BaseSepoliaConstants} from "test/Fork/BaseSepoliaConstants.sol";
+import {L2Resolver} from "src/L2/L2Resolver.sol";
+import {ReverseRegistrar} from "src/L2/ReverseRegistrar.sol";
+
+contract BaseSepoliaForkBase is Test {
+    // RPC alias must be configured in foundry.toml as `base-sepolia`.
+    string internal constant FORK_ALIAS = "base-sepolia";
+
+    // Addresses from constants
+    address internal constant REGISTRY = BaseSepoliaConstants.REGISTRY;
+    address internal constant BASE_REGISTRAR = BaseSepoliaConstants.BASE_REGISTRAR;
+    address internal constant LEGACY_GA_CONTROLLER = BaseSepoliaConstants.LEGACY_GA_CONTROLLER;
+    address internal constant LEGACY_L2_RESOLVER = BaseSepoliaConstants.LEGACY_L2_RESOLVER;
+    address internal constant LEGACY_REVERSE_REGISTRAR = BaseSepoliaConstants.LEGACY_REVERSE_REGISTRAR;
+
+    address internal constant UPGRADEABLE_CONTROLLER_PROXY = BaseSepoliaConstants.UPGRADEABLE_CONTROLLER_PROXY;
+    address internal constant UPGRADEABLE_L2_RESOLVER_PROXY = BaseSepoliaConstants.UPGRADEABLE_L2_RESOLVER_PROXY;
+
+    // ENS L2 Reverse Registrar (Base Sepolia)
+    address internal constant ENS_L2_REVERSE_REGISTRAR = BaseSepoliaConstants.ENS_L2_REVERSE_REGISTRAR;
+
+    // Owners / ops
+    address internal constant L2_OWNER = BaseSepoliaConstants.L2_OWNER;
+
+    // Actors
+    uint256 internal userPk;
+    address internal user;
+
+    // Interfaces
+    RegistrarController internal legacyController;
+    UpgradeableRegistrarController internal upgradeableController;
+    NameResolver internal legacyResolver;
+    IL2ReverseRegistrar internal l2ReverseRegistrar;
+
+    function setUp() public virtual {
+        vm.createSelectFork(FORK_ALIAS);
+
+        // Create a deterministic EOA we control for signing
+        userPk = uint256(keccak256("basenames.fork.user"));
+        user = vm.addr(userPk);
+
+        legacyController = RegistrarController(LEGACY_GA_CONTROLLER);
+        upgradeableController = UpgradeableRegistrarController(UPGRADEABLE_CONTROLLER_PROXY);
+        legacyResolver = NameResolver(LEGACY_L2_RESOLVER);
+        l2ReverseRegistrar = IL2ReverseRegistrar(ENS_L2_REVERSE_REGISTRAR);
+
+        // Ensure legacy resolver authorizes the configured legacy reverse registrar
+        // and ensure the reverse parent node is owned by that registrar so claims succeed.
+        // 1) Ensure ENS owner(BASE_SEPOLIA_REVERSE_NODE) == LEGACY_REVERSE_REGISTRAR
```

Comment:
steps 1 - 4 here are what were necessary to get fork tests passing, and have been translated into an SCM PR for the remaining config